### ISSUE TITLE
UnusedLocalVariable does not support local classes

### ISSUE THREAD

**Issue by Vyom-Yadav** [checkstyle/checkstyle#14092]
I have read check documentation: https://checkstyle.sourceforge.io/checks/coding/unusedlocalvariable.html#UnusedLocalVariable
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
/var/tmp $ javac Test.java
```

```bash
/var/tmp $ cat config.xml
```
```xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
    "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8" />
    <property name="severity" value="warning" />
    <property name="haltOnException" value="false" />
    <module name="TreeWalker">
        <module name="UnusedLocalVariable">
        </module>
    </module>
</module>
```

```bash
/var/tmp $ cat Test.java
```
```java
public class Test {
    int a = 12;

    void foo() {
        int a = 12; // this is unused, no violation
        int ab = 12; // this is unused, violation thrown

        class asd {
            Test a = new Test() {
                void asd() {
                    System.out.println(a);
                }
            };
        }
    }
}
```

```bash
/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-10.12.5-all.jar -c config.xml Test.java
Starting audit...
[WARN] /var/tmp/Test.java:6:9: Unused local variable 'ab'. [UnusedLocalVariable]
Audit done.

```

---

**Describe what you expect in detail.**

I expect a violation to be thrown for both the local variables. The check currently has no logic for scoping local classes, which is the root cause of this false negative. There might be cases where lack of this support might give a false positive.

**Comment by Vyom-Yadav** [comment]
Hey @harshit4311,
You can work on any issue after commenting, "I am on it.". This is a fairly complex issue, and I won't recommend it initially, but if you feel you are comfortable with developing checkstyle, please take it away.

**Comment by aayushRedHat** [comment]
I am onit

**Comment by relentless-pursuit** [comment]
Hello @aayushRedHat, are you still working on the issue? If not, can i try it?

**Comment by aayushRedHat** [comment]
> Hello @aayushRedHat, are you still working on the issue? If not, can i try it?

Hi @relentless-pursuit I am exploring about the internals of this check, I will let you know to pass this issue to you.:)

**Comment by Kevin222004** [comment]
@aayushRedHat are you working on this

**Comment by Lmh-java** [comment]
Hi, is anyone working on this right now? If not, consider I am on it.
